### PR TITLE
NFDIV-4226 Update caseworker general application received event

### PR DESCRIPTION
**Pull request by pallavijustice**
### Change description ###

Update caseworker general application received event to allow adding details of general application.
Re-using pages from solicitor general application event. Moved common ccd pages to common directory.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-4226

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralApplicationType.java [hunk 1]
```diff
@@ -15,14 +15,26 @@ public enum GeneralApplicationType implements HasLabel {
...
+    ORDER_ON_FILLING_OF_ANSWERS("Order on filing of Answers"),
+
+    @JsonProperty("permissionOnDaOot")
+    PERMISSION_ON_DA_OOT("Permission on FO OOT"),
+
+    @JsonProperty("disclosureViaDwp")
+    DISCLOSURE_VIA_DWP("Disclosure via DWP"),
+
+    @JsonProperty("amendApplication")
+    AMEND_APPLICATION("Amend Application"),
```
Comment (adamg-hmcts) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralApplicationType.java:37]: Do we also need to add `Order application without m/c` as a type? I might have missed it

**Review by adamg-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralApplicationReceived.java [hunk 2]
```diff
@@ -16,18 +32,75 @@
...
+            data.getGeneralApplications().add(0, generalApplicationListValue);
+        }
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(data)
+            .state(GeneralApplicationReceived)
+            .build();
+    }
+
+    private PageBuilder addEventConfig(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
```
Comment (DelythJustice) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralApplicationReceived.java:94]: I think we tend to have this defined before the other methods in other classes

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralApplicationType.java (hunk 1 above)
Comment (pallavijustice) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralApplicationType.java:37]: This is similar to Issue divorce without M/C. So consolidated to one option and clarified the final list with Izzy on the jira ticket.

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralApplicationReceived.java [hunk 3]
```diff
@@ -16,18 +32,75 @@
...
+        return new PageBuilder(configBuilder
             .event(CASEWORKER_GENERAL_APPLICATION_RECEIVED)
```
Comment (DelythJustice) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralApplicationReceived.java:96]: Is there a reason for changing the text on this?
I'm wondering if there will be an issue with name change and existing events for history

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralApplicationReceived.java [hunk 4]
```diff
@@ -16,18 +32,75 @@
 import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
 import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
 
+@Slf4j
 @Component
 public class CaseworkerGeneralApplicationReceived implements CCDConfig<CaseData, State, UserRole> {
-    public static final String CASEWORKER_GENERAL_APPLICATION_RECEIVED = "caseworker-general-application-received";
+    public static final String CASEWORKER_GENERAL_APPLICATION_RECEIVED = "cw-general-application-received";
```
Comment (DelythJustice) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralApplicationReceived.java:38]: caseworker to cw ?

Comment (pallavijustice) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralApplicationReceived.java:38]: Had to shorten the name due to issue with the length of event&page combination (70 currently).

**Pull request merged**